docs/developer-docs/integrations/sns/managing/making-proposals.md

```diff
@@ -42,6 +42,619 @@ For these cases, SNSs have so called 'generic proposals'. These are custom propo
+        neuron_grantable_permissions : opt NeuronPermissionList;
+        voting_rewards_parameters : opt VotingRewardsParameters;
+        maturity_modulation_disabled : opt bool;
+        max_number_of_principals_per_neuron : opt nat64;
+    };
+
+    type DefaultFollowees = record { 
+        followees : vec record { 
+            nat64; 
+            Followees 
+        } 
+    };
+
+    type Followees = record { followees : vec NeuronId };
+
+    type NeuronId = record { id : vec nat8 };
+
+    type NeuronPermissionList = record { permissions : vec int32 };
+
+    type VotingRewardsParameters = record {
+        final_reward_rate_basis_points : opt nat64;
+        initial_reward_rate_basis_points : opt nat64;
+        reward_rate_transition_duration_seconds : opt nat64;
+        round_duration_seconds : opt nat64;
+    };
+```
+
+### Putting it together
+
+```candid
+    type ManageNervousSystemParameters: record {
+        default_followees : opt record { 
+            followees : vec record { 
+                        nat64; 
+                        record { followees : vec record { id : vec nat8 } } 
+            } 
+        };
+        max_dissolve_delay_seconds : opt nat64;
+        max_dissolve_delay_bonus_percentage : opt nat64;
+        max_followees_per_function : opt nat64;
+        neuron_claimer_permissions : opt record { permissions : vec int32 };
+        neuron_minimum_stake_e8s : opt nat64;
+        max_neuron_age_for_age_bonus : opt nat64;
+        initial_voting_period_seconds : opt nat64;
+        neuron_minimum_dissolve_delay_to_vote_seconds : opt nat64;
+        reject_cost_e8s : opt nat64;
+        max_proposals_to_keep_per_action : opt nat32;
+        wait_for_quiet_deadline_increase_seconds : opt nat64;
+        max_number_of_neurons : opt nat64;
+        transaction_fee_e8s : opt nat64;
+        max_number_of_proposals_with_ballots : opt nat64;
+        max_age_bonus_percentage : opt nat64;
+        neuron_grantable_permissions : opt record { permissions : vec int32 };
+        voting_rewards_parameters : opt record {
+            final_reward_rate_basis_points : opt nat64;
+            initial_reward_rate_basis_points : opt nat64;
+            reward_rate_transition_duration_seconds : opt nat64;
+            round_duration_seconds : opt nat64;
+        };
+        maturity_modulation_disabled : opt bool;
+        max_number_of_principals_per_neuron : opt nat64;
+    }
+```
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "Lorem ipsum";
+        url = "lorem ipsum";
+        summary = "lorem ipsum";
+        action = opt variant {
+            ManageNervousSystemParameters: record {
+                default_followees : opt record { 
+                    followees : vec record { 
+                                nat64; 
+                                record { followees : vec record { id : vec {1:nat8, 2:nat8} } }  
```

Comment:
Would this compile? I wonder: If I want to set one of these things to a new value, what would be the exact command that I need to use? Can I just plug in a number for nat64? Do I have to fill in all of these things, including those that should not change?
Again, I think a clear example would help

Reply:
I am not certain anything will compile yet, i have been doing these all by hand so far, but yes i intend to make sure all compiles before merging.